src/udev/udev-builtin-path_id.c

```diff
@@ -203,10 +201,25 @@ static sd_device *handle_scsi_sas(sd_device *parent, char **path) {
         }
 
         format_lun_number(parent, &lun);
-        if (sas_address)
+        if (sas_address) {
+                /* Use expander-side phy id */
+                if (sd_device_get_sysname(port, &sysname) < 0)
+                        return NULL;
+                if (!strneq(sysname, "port-", STRLEN("port-")))
```

Comment:
Using `startswith()` function here is simpler.

Reply:
OK, will do.

Reply:
Note: `startswith` returns the pointer to the rest of the string. Please use it in the following line.